tests/unit/tf/models/test_retrieval.py

```diff
@@ -515,6 +515,36 @@ def test_two_tower_model_with_custom_options(
     )
 
 
+def test_two_tower_v2_with_sample_weight(ecommerce_data: Dataset):
```

Comment:
I think it would be good using the binary cross entropy loss in the test, so that it serve as an example